### PR TITLE
Fix: Redis plugin: extract cmd and args based on regex

### LINKED ISSUE #6363: [Bug] Redis plugin: handle space in string args
## Description
Currently Redis plugin splits the overall string by whitespace and assumes first token to be cmd, and rest to be args. However, this fails in case a string arg is present which has spaces in it. 

Context:
https://discord.com/channels/725602949748752515/760761686549463060/872097304734224414

### PR DESCRIPTION
**Pull request by sumitsum**
## Description
- Earlier `split` method was used to segregate cmd and args which failed when multiple words inside a quoted string formed one argument. e.g. `set key "my value"` would produce `set`, `key`, `"my`, `value"` as the tokens when `split` method is used, which is not correct. This change introduces a regex that would create the following tokens: `set`, `key`, `"my value"`.

Fixes #6363 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- JUnit TC
- Manual

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-plugins/redisPlugin/src/main/java/com/external/plugins/RedisPlugin.java [hunk 1]
```diff
@@ -68,21 +73,29 @@ public RedisPlugin(PluginWrapper wrapper) {
...
 
-                // First value will be the redis command and others are arguments for that command
-                String[] bodySplitted = query.trim().split("\\s+");
+                Map cmdAndArgs = getCommandAndArgs(query.trim());
+                if (!isQueryFormatValid(cmdAndArgs)) {
+                    return Mono.error(
+                            new AppsmithPluginException(
+                                    AppsmithPluginError.PLUGIN_EXECUTE_ARGUMENT_ERROR,
+                                    "Appsmith server has failed to parse Redis query. Please check if the Redis query" +
+                                            " has correct format."
```
Comment (mohanarpit) [inline app/server/appsmith-plugins/redisPlugin/src/main/java/com/external/plugins/RedisPlugin.java:82]: Nit: Minor English grammar correction
```suggestion
                                    "Appsmith server has failed to parse your Redis query. Are you sure it's" +
                                            " been formatted correctly."
```

File: app/server/appsmith-plugins/redisPlugin/src/main/java/com/external/plugins/RedisPlugin.java [hunk 2]
```diff
@@ -68,21 +73,29 @@ public RedisPlugin(PluginWrapper wrapper) {
...
 
                 Protocol.Command command;
                 try {
                     // Commands are in upper case
-                    command = Protocol.Command.valueOf(bodySplitted[0].toUpperCase());
+                    command = Protocol.Command.valueOf((String) cmdAndArgs.get(CMD_KEY));
                 } catch (IllegalArgumentException exc) {
                     return Mono.error(new AppsmithPluginException(AppsmithPluginError.PLUGIN_EXECUTE_ARGUMENT_ERROR,
-                            String.format("Not a valid Redis command:%s", bodySplitted[0])));
+                            String.format("Not a valid Redis command:%s", cmdAndArgs.get(CMD_KEY))));
```
Comment (mohanarpit) [inline app/server/appsmith-plugins/redisPlugin/src/main/java/com/external/plugins/RedisPlugin.java:93]:
```suggestion
                            String.format("Not a valid Redis command: %s", cmdAndArgs.get(CMD_KEY))));
```

File: app/server/appsmith-plugins/redisPlugin/src/main/java/com/external/plugins/RedisPlugin.java [hunk 3]
```diff
@@ -125,6 +138,43 @@ public RedisPlugin(PluginWrapper wrapper) {
                     .subscribeOn(scheduler);
         }
 
+        private boolean isQueryFormatValid(Map cmdAndArgs) {
+            if (cmdAndArgs.containsKey(CMD_KEY)) {
+                return true;
+            }
+
+            return false;
```
Comment (mohanarpit) [inline app/server/appsmith-plugins/redisPlugin/src/main/java/com/external/plugins/RedisPlugin.java:146]: Nit: This function can be simplified into a single line statement.
```suggestion
            return cmdAndArgs.containsKey(CMD_KEY);
```

File: app/server/appsmith-plugins/redisPlugin/src/test/java/com/external/plugins/RedisPluginTest.java [hunk 4]
```diff
@@ -243,7 +243,7 @@ public void itShouldExecuteCommandWithArgs() {
 
         // Setting a key
         ActionConfiguration setActionConfiguration = new ActionConfiguration();
-        setActionConfiguration.setBody("SET key value");
+        setActionConfiguration.setBody("SET key \"my value\"");
```
Comment (mohanarpit) [inline app/server/appsmith-plugins/redisPlugin/src/test/java/com/external/plugins/RedisPluginTest.java:246]: Is it possible to have multiple tests with & without quotes? What about single quotes? Are those valid Redis commands as well? Do we need to handle them as well?

File: app/server/appsmith-plugins/redisPlugin/src/main/java/com/external/plugins/RedisPlugin.java [hunk 5]
```diff
@@ -125,6 +138,43 @@ public RedisPlugin(PluginWrapper wrapper) {
...
+
+        private Map getCommandAndArgs(String query) {
+            /**
+             * - This regex matches either a whole word, or anything inside double quotes.
+             * - e.g. if the query string is: set key "my val", then the regex matches the following:
+             * (1) set
+             * (2) key
+             * (3) "my val"
+             */
+            String redisCmdRegex = "\\\"[^\\\"]+\\\"|[\\S]+";
```
Comment (nidhi-nair) [inline app/server/appsmith-plugins/redisPlugin/src/main/java/com/external/plugins/RedisPlugin.java:157]: Are single quotes also valid in redis?

File: app/server/appsmith-plugins/redisPlugin/src/main/java/com/external/plugins/RedisPlugin.java [hunk 6]
```diff
@@ -125,6 +138,43 @@ public RedisPlugin(PluginWrapper wrapper) {
                     .subscribeOn(scheduler);
         }
 
+        private boolean isQueryFormatValid(Map cmdAndArgs) {
```
Comment (nidhi-nair) [inline app/server/appsmith-plugins/redisPlugin/src/main/java/com/external/plugins/RedisPlugin.java:141]: This method seems a little redundant, can we make the check inline?

File: app/server/appsmith-plugins/redisPlugin/src/test/java/com/external/plugins/RedisPluginTest.java (hunk 4 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/redisPlugin/src/test/java/com/external/plugins/RedisPluginTest.java]: Thanks for pointing this out. I have updated the regex to handle single quotes as well. I have updated the TC will 6 different types of inputs with a mix of double, single and no quotes.

File: app/server/appsmith-plugins/redisPlugin/src/main/java/com/external/plugins/RedisPlugin.java (hunk 5 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/redisPlugin/src/main/java/com/external/plugins/RedisPlugin.java]: Thanks for pointing this out. I have updated the regex to handle single quotes as well. I have updated the TC will 6 different types of inputs with a mix of double, single and no quotes.

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**